Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:latest

```

Comment:
Hi again, as I was a bit rusty on Docker, I consulted with experts and they gave me this feedback about this dockerfile:

- A "fixed" version of the base image should be used, e.g., ubuntu:22.04. latest will change with new Ubuntu releases (which will impact the versions of the installed dependencies)
- Best practice is to reduce the number of layers. In particular all the apt-get command should be one command: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get. We should also combine all the build commands (mkdir, cmake, make) into one.
- If the image were to be distributed a mutli-stage Dockerfile should be used, i.e., the output image does not require all the -dev dependencies to run => results in a smaller output image
- The last cd command is kind of useless; the CMD statement later uses all absolute paths. Also, it would be more efficient to use WORKDIR in this case (does not create a layer)
- The ARG command is probably incorrect. If the intention was to set an ENV var ENV should be used. In any case, it can be added to the RUN command without having to set it using ARG/ENV
- Instead of cloning the repository, COPY should be used COPY . /fast_methods/ makes more sense and it is safer.
- We should include a .dockerignore (same content as .gitignore) to exclude the build directory, etc. from being passed in as part of the build context (otherwise, the mkdir build command might actually fail when using COPY
- Also, please add a sentence in the README about the Docker support (how to build; how to use)

Reply:
Ok. I'm a noob with Docker also. I'll try my best.